manifests/ca.pp

```diff
@@ -1,11 +1,39 @@
+# @summary
+#  Creates per CA configuration files.
+#
+# @example
+#  fetchcrl::ca{'EDG-Tutorial-CA':
+#    agingtolerance => 168
+#  }
+#
+# @param name
+#  The name of the CA to manage a configuration for.
+#
+# @param anchorname
+#  The name of the CA to manage a configuration for.
+#
+# @param nowarnings
+#  Should warnings be supressed for this CA.
+#
+# @param noerrors
+#  Should errors be supressed for this CA.
+#
+# @param httptimeout
+#  The timeout for this CA.
+#
+# @param agingtolerance
+#  The delay if failures before it is considered an error.
+#
+# @param crl_url
+#  A list of URLs to download CAs from
 #
 define fetchcrl::ca(
-  $anchorname     = $title,
-  $nowarnings     = true,
-  $noerrors       = true,
-  $httptimeout    = undef,
-  $agingtolerance = undef,
-  $crl_url        = []
+  String[1] $anchorname = $title,
+  Boolean $nowarnings                       = false,
+  Boolean $noerrors                         = false,
+  Optional[Integer] $httptimeout            = undef,
+  Optional[Integer] $agingtolerance         = undef,
+  Optional[Array[Stdlib::Httpurl]] $crl_url = [],
```

Comment:
Isn't `Optional` redundant when there's a default?

Reply:
I think only if the default is an Integer. It's saying Interger or unset.

Reply:
Another way of writing `Optional` is `Variant[Undef, Array[Stdlib::Httpurl]]`. Since a default means you can't really set it to `undef` anymore, I think you can drop `Optional`.

Reply:
But yes that optional on the Array is totally pointless indeed - thanks.

Reply:
For completeness; I just gave that variant to illustrate. I'm perfectly happy with `Optional` when it's appropriate :)